modules/oe_translation_cdt/tests/src/Kernel/CdtParametersCompilerPassTest.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\Tests\oe_translation_cdt\Kernel;
+
+use Drupal\Core\DependencyInjection\ContainerBuilder;
+use Drupal\Core\Site\Settings;
+use Drupal\oe_translation_cdt\Compiler\CdtParametersCompilerPass;
+use Drupal\Tests\oe_translation\Kernel\TranslationKernelTestBase;
+
+/**
+ * Tests the compiler pass that sets CDT configuration.
+ *
+ * @group batch1
+ */
+class CdtParametersCompilerPassTest extends TranslationKernelTestBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected static $modules = [
+    'oe_translation_cdt',
+    'oe_translation_remote',
+  ];
+
+  /**
+   * Tests that the CDT configuration is passed properly to the container.
+   */
+  public function testCdtCompilerPass(): void {
+    // Define strings with "%" characters, that need to be escaped.
+    $settings = [
+      'cdt.base_api_url' => '%url%',
+      'cdt.username' => '%username',
+      'cdt.client' => '%%client%%',
+      'cdt.password' => 'test_%%password%',
+      'cdt.api_key' => 'test_%api_key',
+    ];
+    new Settings($settings);
+
+    // Define a simple inline class that will receive the parameters.
+    $receiver_object = new class {
```

Comment:
I didn't use the original `ApiClient` because I'd have to make its properties available via the reflection. And this would be hard to maintain if anything changed on the cdt-client side. This minimalistic approach seems reasonable.